src/Meshes/Builders/goldbergBuilder.ts

```diff
@@ -0,0 +1,369 @@
+        const ba = b.subtract(a);
+        const ca = c.subtract(a);
+        const norm = Vector3.Cross(ca, ba).normalize();
+        for (let v = 0; v < verts.length; v++) {
+            normals.push(norm.x, norm.y, norm.z);
+            const pdata = goldbergData.vertex[verts[v]];
+            positions.push(pdata[0] * sizeX, pdata[1] * sizeY, pdata[2] * sizeZ);
+            uvs.push((pdata[0] * sizeX - minX)/(maxX - minX), (pdata[1] * sizeY - minY)/(maxY - minY));
+        }
+        for (let v = 0; v < verts.length - 2; v++) {
+            indices.push(index, index + v + 2, index + v + 1);
+        }
+        index += verts.length;
+    }
+
+    VertexData._ComputeSides(sideOrientation, positions, indices, normals, uvs, options.frontUVs, options.backUVs);
+
+    const vertexData = new VertexData();
+    vertexData.positions = positions;
+    vertexData.indices = indices;
+    vertexData.normals = normals;
+    vertexData.uvs = uvs;
+    return vertexData;
+};
+
+/**
+ * Class adding properties and methods to a Goldberg Mesh
+ * @hidden
+ */
+export class GDMesh extends Mesh {
+
+    constructor(
+        name: string,
+        scene: Nullable<Scene> = null,
+        geoData: GeoData,
+        goldbergData: PolyhedronData
+    ) {
+        super(name, scene);
+
+        scene = this.getScene();
+
+        this._nbSharedFaces = geoData._sharedNodes;
+        this._nbUnsharedFaces = geoData._poleNodes;
+        this._adjacentFaces = geoData._adjacentFaces;
+        this._nbFaces = this._nbSharedFaces + this._nbUnsharedFaces;
+        this._nbFacesAtPole = (this._nbUnsharedFaces - 12) / 12;
+        for (let f = 0; f < geoData.vertex.length; f++) {
+            this._faceCenters.push(Vector3.FromArray(geoData.vertex[f]));
+            this._faceColors.push(new Color4(1, 1, 1, 1));
+        };
+        for (let f = 0; f < goldbergData.face.length; f++) {
+            const verts = goldbergData.face[f];
+            const a = Vector3.FromArray(goldbergData.vertex[verts[0]]);
+            const b = Vector3.FromArray(goldbergData.vertex[verts[2]]);
+            const c = Vector3.FromArray(goldbergData.vertex[verts[1]]);
+            const ba = b.subtract(a);
+            const ca = c.subtract(a);
+            const norm = Vector3.Cross(ca, ba).normalize();
+            const z = Vector3.Cross(ca, norm).normalize();
+            this._faceXaxis.push(ca.normalize());
+            this._faceYaxis.push(norm);
+            this._faceZaxis.push(z);
+        };
+        this._setMetadata();
+    }
+
+    private _faceColors: Color4[] = [];  
+    private _faceCenters: Vector3[] = [];
+    public _faceZaxis: Vector3[] = [];
+    public _faceXaxis: Vector3[] = [];
+    public _faceYaxis: Vector3[] = [];
+    private _nbSharedFaces: number;
+    private _nbUnsharedFaces: number;
+    private _nbFaces: number;
+    private _nbFacesAtPole: number;
+    private _adjacentFaces: number[][];
+
+    private _setMetadata = () => {

```

Comment:
Why not declaring this function as usual?
```typescript
private _setMetadata() {
...
}
```

Same comment for all functions starting at `refreshFaceData` below.

The user can't overload those properties as the class is hidden, so I would advise to declare the functions as usual.

Reply:
There is a bigger problem with refreshFaceData since it was intended to reset properties from the metadata after a Goldberg mesh was imported but ... see 
https://github.com/BabylonJS/Babylon.js/pull/11036#issuecomment-916865324